src/libs/SubscriptionUtils.ts

```diff
@@ -0,0 +1,133 @@
+import {differenceInCalendarDays, fromUnixTime, isAfter, isBefore, parse as parseDate} from 'date-fns';
+import type {OnyxCollection, OnyxEntry} from 'react-native-onyx';
+import Onyx from 'react-native-onyx';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {BillingGraceEndPeriod, Policy} from '@src/types/onyx';
+
+let firstDayFreeTrial: OnyxEntry<string>;
+Onyx.connect({
+    key: ONYXKEYS.NVP_FIRST_DAY_FREE_TRIAL,
+    callback: (value) => (firstDayFreeTrial = value),
+});
+
+let lastDayFreeTrial: OnyxEntry<string>;
+Onyx.connect({
+    key: ONYXKEYS.NVP_LAST_DAY_FREE_TRIAL,
+    callback: (value) => (lastDayFreeTrial = value),
+});
+
+let userBillingFundID: OnyxEntry<number>;
+Onyx.connect({
+    key: ONYXKEYS.NVP_BILLING_FUND_ID,
+    callback: (value) => (userBillingFundID = value),
+});
+
+let userBillingGraceEndPeriodCollection: OnyxCollection<BillingGraceEndPeriod>;
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.SHARED_NVP_PRIVATE_USER_BILLING_GRACE_PERIOD_END,
+    callback: (value) => (userBillingGraceEndPeriodCollection = value),
+    waitForCollectionCallback: true,
+});
+
+let ownerBillingGraceEndPeriod: OnyxEntry<number>;
+Onyx.connect({
+    key: ONYXKEYS.NVP_PRIVATE_OWNER_BILLING_GRACE_PERIOD_END,
+    callback: (value) => (ownerBillingGraceEndPeriod = value),
+});
+
+let amountOwed: OnyxEntry<number>;
+Onyx.connect({
+    key: ONYXKEYS.NVP_PRIVATE_AMOUNT_OWNED,
+    callback: (value) => (amountOwed = value),
+});
+
+let allPolicies: OnyxCollection<Policy>;
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.POLICY,
+    callback: (value) => (allPolicies = value),
+    waitForCollectionCallback: true,
+});
+
+/**
+ * Calculates the remaining number of days of the workspace owner's free trial before it ends.
+ */
+function calculateRemainingFreeTrialDays(): number {
+    if (!lastDayFreeTrial) {
+        return 0;
+    }
+
+    const currentDate = new Date();
+    const difference = differenceInCalendarDays(parseDate(lastDayFreeTrial, CONST.DATE.FNS_DATE_TIME_FORMAT_STRING, currentDate), currentDate);
```

Comment:
Would not `difference`  become `0` when the `currentDate` is `lastDayFreeTrial` and be considered as trial ended even when it has not really ended?

Reply:
Adjusted the logic to calculate with second precision cc @chiragsalian